src/code-editor/resizable-box/index.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useEffect, useRef, useState } from 'react';
+import { useStableEventHandler } from '../../internal/hooks/use-stable-event-handler';
+import styles from './styles.css.js';
+
+export interface ResizeBoxProps {
+  children: React.ReactNode;
+  height: number;
+  minHeight: number;
+  onResize: (newHeight: number) => void;
+}
+
+export function ResizableBox({ children, height, minHeight, onResize }: ResizeBoxProps) {
+  const [dragOffset, setDragOffset] = useState<null | number>(null);
+  const onResizeStable = useStableEventHandler(onResize);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const onMouseDown: React.MouseEventHandler = event => {
+    if (event.button !== 0 || !containerRef.current) {
+      return;
+    }
+    const containerBottom = containerRef.current.getBoundingClientRect().bottom;
+    setDragOffset(containerBottom - event.clientY);
+  };
+
+  useEffect(() => {
+    if (dragOffset === null || !containerRef.current) {
+      return;
+    }
+    const container = containerRef.current;
+
+    const onMouseMove = (event: MouseEvent) => {
+      const { top } = container.getBoundingClientRect();
+      const cursor = event.clientY;
+      onResizeStable(Math.max(cursor + dragOffset - top, minHeight));
+    };
+    const onMouseUp = () => {
+      setDragOffset(null);
+    };
+    document.body.classList.add(styles['resize-active']);
+    document.addEventListener('mousemove', onMouseMove);
+    document.addEventListener('mouseup', onMouseUp);
```

Comment:
Is this `mouseup` event also fired when the mouse button is released outside of the browser window / document? I've encountered implementations where the resizing becomes "sticky" to the mouse if you release the button outside. We might need to listen on `window` instead or use something like [`setPointerCapture()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture)

Reply:
1. I have copied this code from [resizable table columns implementation](https://github.com/cloudscape-design/components/blob/main/src/table/resizer/index.tsx). We have not received any related bug reports there. (Why not reusing the code between these two? Because resizing tables is much different than divs, very few lines to share)
2. Regardless of the above, tested locally, it stops the resize as expected

Reply:
Great, thank you for verifying!